Jenkinsfile

```diff
@@ -145,6 +161,44 @@ pipeline {
                 }
             }
         }
+        stage('Verify changes') {
+            agent any
+            steps{
+                script {         
+
+                    retry(3) { checkout scm }
+                    sh "git clean -xffd"
+
+                    def commitHash = sh(script: "git rev-parse HEAD | tr '\\n' ' '", returnStdout: true)
+                    sh(script: "git pull && git checkout develop", returnStdout: false)
+
+                    def bashScript = """
+                        for i in ${scPaths};
+                        do
+                            git diff ${commitHash}develop -- \$i
+                        done
+                    """
+
+                    def differences = sh(script: bashScript, returnStdout: true)
+
+                    println differences
+
+                    if (differences?.trim()) {
+                        println "There are differences in the source code, CI/CD will run."
+                        skipRemainingStages = false
+                    }
+                    else{
+                        println "There aren't any differences in the source code, CI/CD will not run."
+                        skipRemainingStages = true
+                    }
+                }
+            }
+            post {
+                always {
+                    deleteDir()
+                }
+            }
+        }
         stage('Headers checks') {
```

Comment:
Can we also not run headers checks if there are no changes? I think there is no reason to run the headers check here.